src/app/components/recipe-page/recipe-page.component.ts

```diff
@@ -1,45 +1,46 @@
 import { Component, OnInit } from '@angular/core';
-import { ApiService } from "./../api.service";
+import { ApiService } from './../api.service';
 import { Recipe } from './../models/recipe.model';
 import { ActivatedRoute, Router } from '@angular/router';
-import { switchMap } from 'rxjs/operators';
-
-
-
 
 @Component({
   selector: 'app-recipe-page',
   templateUrl: './recipe-page.component.html',
-  styleUrls: ['./recipe-page.component.css']
+  styleUrls: ['./recipe-page.component.css'],
 })
-
-
-
 export class RecipePageComponent implements OnInit {
   recipe: Recipe;
+  public edit: boolean;
   public id: Number;
-  constructor(private recipesService: ApiService,
+  constructor(
+    private recipesService: ApiService,
     private route: ActivatedRoute,
     private router: Router
-    ) {
-    // this.recipe = ;
-  }
+  ) {}
 
   ngOnInit(): void {
+    this.edit = false;
     this.id = parseInt(this.route.snapshot.paramMap.get('id'));
-    this.recipesService.getRecipe(this.id).subscribe((recipe: any) =>
-    {
-      this.recipe = recipe;
-      console.log(recipe)
-    })
+    this.getRecipe();
   }
 
-  handleDelete() {
-    console.log(`handling delete: ${this.id}`)
-    this.recipesService.deleteRecipe(this.id).toPromise()
-    .then(() => alert("Deleted!"))
-    .then(() => this.router.navigate(['/']));
+  getRecipe(): void {
+    this.recipesService.getRecipe(this.id).subscribe((recipe: Recipe) => {
```

Comment:
Then here use `.toPromise()` instead of `subscribe()` and use async await. Verify it though, because I'm not 100% sure that it will work this way.